app/scripts/modules/core/src/entityTag/alerts/CategorizedNotifications.tsx

```diff
@@ -0,0 +1,124 @@
+ * |                  | notification 2                    |
+ * |                  |                                   |
+ * |                  |                                   |
+ * |                  |                                   |
+ * +------------------+-----------------------------------+
+ */
+@autoBindMethods
+export class CategorizedNotifications extends React.Component<IProps, IState> {
+  public state: IState;
+
+  constructor(props: IProps) {
+    super(props);
+
+    const categorizedAlerts = this.categorizeNotifications(props.notifications);
+    this.state = {
+      categorizedAlerts,
+      selectedCategory: categorizedAlerts[0].category,
+    };
+  }
+
+  public componentWillReceiveProps(props: IProps): void {
+    const categorizedAlerts = this.categorizeNotifications(props.notifications);
+    this.setState({ categorizedAlerts });
+  }
+
+  /**
+   * Given a list of alerts, groups into categories (entityTag namespaces, e.g., 'deprecation')
+   */
+  private categorizeNotifications(allNotifications: INotification[]): ICategoryNotifications[] {
+    const getCategory = (notification: INotification) => notification.entityTag.category || 'default';
+
+    const makeAlertCategory = (name: string): ICategoryNotifications => {
+      const category = NotificationCategories.getCategory(name);
+
+      return {
+        category,
+        notifications: allNotifications.filter((n: INotification) => getCategory(n) === name),
+      };
+    };
+
+    // unique category names found in the alerts.
+    const categories: string[] = uniq(allNotifications.map(getCategory));
+    return categories.map(makeAlertCategory).sort((a, b) => b.category.severity - a.category.severity);
+  }
+
+  private handleCategorySelected(selectedCategory: INotificationCategory): void {
+    this.setState({ selectedCategory });
+  }
+
+  public render() {
+    const { categorizedAlerts, selectedCategory } = this.state;
+    const { onEditTag, onDeleteTag } = this.props;
+    if (!categorizedAlerts || !selectedCategory) {
+      return null;
+    }
+
+    const selectedNotifications: INotification[] = categorizedAlerts.find(tuple => tuple.category === selectedCategory).notifications;
+
+    return (
+      <div className="flex-container-h">
+        <ul className="list-group notification-categories">
+          { categorizedAlerts.map(({ category, notifications }) => (
+              <NotificationCategory
+                isSelected={category === selectedCategory}
+                key={category.id}
+                category={category}
+                notifications={notifications}
+                onCategorySelected={this.handleCategorySelected}
+              />
+            )
+          )}
+        </ul>
+
+        { this.props.grouped ?
+          <GroupedNotificationList notifications={selectedNotifications} /> :
+          <NotificationList notifications={selectedNotifications} onEditTag={onEditTag} onDeleteTag={onDeleteTag} />
```

Comment:
Render the list of notifications. If `grouped` is true, then group and collapse duplicated notification messages for multiple server groups